libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -2096,7 +2096,7 @@ MQTTStatus_t MQTT_DeserializePublish( const MQTTPacketInfo_t * pIncomingPacket,
     }
     else if( ( pIncomingPacket->type & 0xF0U ) != MQTT_PACKET_TYPE_PUBLISH )
     {
-        LogError( ( "Packet is not publish. Packet type: %hu.",
+        LogError( ( "Packet is not publish. Packet type: %02x.",
```

Comment:
If this is being logged in hexadecimal, should it have the `0x` prefix?
```suggestion
        LogError( ( "Packet is not publish. Packet type: 0x%02x.",
```

Reply:
Maybe, right now it's consistent with how we log the packet type everywhere else. If we added `0x` we'd want to do it everywhere, which might be better for another PR. I only changed it because I got a build warning due to the format specifying an `unsigned short` but the type was a `uint8_t`